testnet/stacks-node/src/neon_node.rs

```diff
@@ -4182,6 +4174,17 @@ impl StacksNode {
         globals.set_initial_leader_key_registration_state(leader_key_registration_state);
 
         let relayer_thread = RelayerThread::new(runloop, local_peer.clone(), relayer);
+
+        let public_key = keychain.get_pub_key();
+        let miner_addr = relayer_thread.bitcoin_controller.get_miner_address(StacksEpochId::Epoch21, &public_key);
```

Comment:
I discussed this with @kantai, and he felt that hard-coding this parameter to Epoch21 made sense - want to chime in?

Reply:
Yeah, so the issue here is that the "miner address" in the bitcoin controller behaves somewhat surprisingly with epochs and segwit configuration:

```rust
        if self.config.miner.segwit && epoch_id >= StacksEpochId::Epoch21 {
            let hash160 = Hash160::from_data(&public_key.to_bytes_compressed());
            BitcoinAddress::from_bytes_segwit_p2wpkh(network_id, &hash160.0)
                .expect("Public key incorrect")
        } else {
            let hash160 = Hash160::from_data(&public_key.to_bytes());
            BitcoinAddress::from_bytes_legacy(
                network_id,
                LegacyBitcoinAddressType::PublicKeyHash,
                &hash160.0,
            )
            .expect("Public key incorrect")
        }
```

So, if `segwit = true`, then the miner mines from _two different bitcoin addresses_ if the epoch boundary is crossed.

As it pertains to the mining monitoring (which is where this variable is used), the monitoring thread assumes a globally static miner address. That's not actually the case, unfortunately, so this is kind of a hack -- if the miner config setting opts into segwit mining, the monitoring code always expects to be watching for a segwit mining address, even if epoch-2.1 hasn't started yet.

I think this is okay in the monitoring code, but I do think it's bad code hygiene, and should have a comment reflecting this logic (and be separated away from the rest of the spawn code into its own method). I think that really the `get_miner_address` behavior should be changed so that it no longer accepts an epoch parameter, but I understand that probably impacts the integration tests (I cannot imagine a real world miner at any point depended on this code -- it would have required preemptively funding the segwit address for the switch over).